pkg/bytecode/value.go

```diff
@@ -134,6 +151,14 @@ func (a arrayVal) Index(idx value) (value, error) {
 	return a.Elements[index], nil
 }
 
+func (a arrayVal) Slice(start, end value) (value, error) {
+	startIdx, endIdx, err := normalizeSliceIndices(start, end, len(a.Elements))
+	if err != nil {
+		return nil, err
+	}
+	return arrayVal{Elements: a.Elements[startIdx:endIdx]}, nil
```

Comment:
This is not copying the slice so mutating the original slice will mutate the slice var and vice versa:

From https://docs.evy.dev/spec.html#index-and-slice : 
> The slice expression `arr[start:end]` copies a substring or subarray [...]

Evaluator implementation: https://github.com/evylang/evy/blob/main/pkg/evaluator/value.go#L232

As commented on the next commit, the test there for testing this mutation isolation does not look like it works due to index assignments not being correctly implemented.

Reply:
My bad, I guess the assignment problems obfuscated the real problem for me. I will revisit once the new PR is pushed to support assigning to index expressions.